src/groupByKey.ts

```diff
@@ -2,6 +2,26 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  const obj: GroupsMap<T> = {};
+
+  items.forEach((item: T) => {
+    const newKey: T[keyof T] = item[key];
```

Comment:
```suggestion
    const newKey = item[key];
```
redundant. ts will now type of newKey as items and key already typed.